core/commands/IssueAddCommand.php

```diff
@@ -219,9 +219,9 @@ function validate() {
 
 		if( isset( $t_issue['tags'] ) && is_array( $t_issue['tags'] ) && !tag_can_create( $this->user_id ) ) {
 			foreach( $t_issue['tags'] as $t_tag ) {
-				if( $t_tag['id'] === -1 ) {
+				if( isset( $t_tag['id'] ) && $t_tag['id'] === -1 ) {
```

Comment:
We need to implement a method that takes a tag element and return an id or -1 if not found.

- If id supplied - verify it exists and return it.
- If name supplied (case insensitive), lookup the name and return its id.

The method can be designed to throw Client exception with appropriate error message based on what is supplied, (e.g. id, name, id + name, none).  The method should be used during validation and execution.

A similar, but not exactly the same (doesn't throw and used by SOAP hence transforms objects to array) method is `mci_get_project_id`.